### PR TITLE
Gcs related Feature files and step design

### PR DESCRIPTION
**Pull request by sumitsri12**
This PR needs to be reviewed by internal Cloudsufi team only
@sumitnigamcs

### CODE REVIEW

File: src/e2e-test/features/GCSRuntime.feature [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    Then Enter the GCS Properties with GCS bucket "gcsCsvbucket" and format "gcsCSVFileFormat"
+    Then Close the GCS Properties
+    Then Enter the BigQuery Properties for the table "gcsBqTableName"
+    Then Close the BigQuery Properties
+    Then Save and Deploy Pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Verify the pipeline status in each case
+    Then Open Logs
+#    Then validate failedMessage is displayed
```
Comment (sumitnigamcs) [inline src/e2e-test/features/GCSRuntime.feature]: remove this commented line

File: src/e2e-test/features/GCSRuntime.feature [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    Then Enter the GCS Properties with GCS bucket "gcsTextbucket" and format "gcsTextFileFormat"
+    Then Close the GCS Properties
+    Then Enter the BigQuery Properties for the table "gcsBqTableName"
+    Then Close the BigQuery Properties
+    Then Save and Deploy Pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Verify the pipeline status in each case
+    Then Open Logs
+    Then validate successMessage is displayed
```
Comment (sumitnigamcs) [inline src/e2e-test/features/GCSRuntime.feature]: have you executed mvn verify command? look like some checkstyle issues are there

File: src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSConnector.java [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    wait.until(ExpectedConditions.
+                 invisibilityOfElementLocated(By.xpath("//h5[contains(text(),'Browse Connections')]")));
+    CdfGcsActions.enterSamplesize();
+    CdfGcsActions.getGcsBucket(CdapUtils.pluginProp(bucket));
+    CdfGcsActions.selectFormat(CdapUtils.pluginProp(format));
+    CdfGcsActions.skipHeader();
+    CdfGcsActions.enterMaxsplitsize(CdapUtils.pluginProp("gcsMaxsplitSize"));
+    CdfGcsActions.enterMinsplitsize(CdapUtils.pluginProp("gcsMinsplitSize"));
+    CdfGcsActions.enterRegexpath(CdapUtils.pluginProp("gcsRegexpath"));
+    CdfGcsActions.enterPathfield(CdapUtils.pluginProp("gcspathField"));
```
Comment (sumitnigamcs) [inline src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSConnector.java]: correct the camelCasing.....go through Ragini's PR and check my comments

File: src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/package-info.java [hunk 4]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Package contains the stepDesign for the GCP features.
+ */
+package io.cdap.plugin.gcs.stepsdesign;
+
+
```
Comment (sumitnigamcs) [inline src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/package-info.java:6]: why new line here

**Review by sumitnigamcs — CHANGES_REQUESTED** [review]

File: src/e2e-test/features/GCSError.feature [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  Then Verify reference name is mandatory
+
+  @TC-Error-2
+  Scenario: Verify Error validation in path field
+    Given Open Datafusion Project to configure pipeline
+    When Source is GCS bucket
+    When Target is BigQuery
+    Then Link Source and Sink to establish connection
+    Then Enter the GCS Properties with format "gcsCSVFileFormat" by entering blank path
+    Then Verify path is mandatory
```
Comment (sawantpritam) [inline src/e2e-test/features/GCSError.feature]: No newline at end of file.

File: src/e2e-test/features/GCSRuntime.feature [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    Then Enter the GCS Properties with GCS bucket "gcsTextbucket" and format "gcsTextFileFormat"
+    Then Close the GCS Properties
+    Then Enter the BigQuery Properties for the table "gcsBqTableName"
+    Then Close the BigQuery Properties
+    Then Save and Deploy Pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Verify the pipeline status is "Succeeded"
+    Then Open Logs
+    Then Validate successMessage is displayed
```
Comment (sawantpritam) [inline src/e2e-test/features/GCSRuntime.feature:87]: No newline at end of file.

File: src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSConnector.java [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} by entering blank referenceName")
+  public void enterTheGCSPropertiesByPuttingBlankValueInMandatoryFields(String bucket, String format)
+    throws IOException, InterruptedException {
+    CdfGcsActions.gcsProperties();
+    CdfGcsActions.getGcsBucket(CdapUtils.pluginProp(bucket));
+    CdfGcsActions.selectFormat(CdapUtils.pluginProp(format));
+  }
+
+  @Then("verify the schema in output")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSConnector.java]: verify -> Verify

File: src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSConnector.java [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+  public void verifyTheSchemaInOutput() {
+    CdfGcsActions.validateSchema();
+  }
+
+  @Then("verify the datatype")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSConnector.java:103]: verify -> Verify

File: src/e2e-test/resources/errorMessage.properties [hunk 9]
```diff
@@ -0,0 +1,5 @@
+errorMessageTemporaryBucket=Bucket name can only contain lowercase letters, numbers, '.', '_', and '-'.
+errorMessageDataset=Required property 'dataset' has no value.
+errorMessageReference=Required property 'referenceName' has no value.
+errorMessageColor=#a40403
+errorMessagePath=Required property 'path' has no value.
```
Comment (sawantpritam) [inline src/e2e-test/resources/errorMessage.properties]: No newline at end of file.

**Review by sawantpritam — CHANGES_REQUESTED** [review]

**Pull request closed without merging**